### PR TITLE
feat: add script runner

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

https://github.com/orgs/ApeWorX/projects/1#card-54989179

TODO:
- [ ] Pass context variables to `cli.Group`/`cli.Command` invocation

### CODE REVIEW

File: src/ape_run/__init__.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+from ape.utils import Abort
+
+
+def _run_script(script_file, interactive=False):
+    # Load the python module and find our hook functions
+    ns: dict = {}
+    try:
+        py_module = compile(script_file.read_text(), str(script_file), "exec")
+        eval(py_module, ns, ns)
```
Comment (lost-theory) [inline src/ape_run/__init__.py:13]: Not sure what the tradeoffs would be but I think you can also do this step using `importlib`:

https://docs.python.org/3/library/importlib.html

**Review by lost-theory — APPROVED** [review]
LGTM. Left one minor observation.

File: src/ape_run/__init__.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape_run/__init__.py:13]: `importlib` seems clearer!

**Review by lost-theory — APPROVED** [review]
LGTM.

File: src/ape_run/_cli.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def _run_script(script_file, interactive=False):
+    # Load the python module and find our hook functions
+    import_str = ".".join(script_file.parts[:-1] + (script_file.stem,))
+    try:
+        py_module = import_module(import_str)
+
+    except Exception as e:
+        raise Abort(e.with_traceback()) from e
```
Comment (lost-theory) [inline src/ape_run/_cli.py:20]: I think you want:

```
raise Abort().with_traceback(e.__traceback__)
```

Or:

```
raise Abort() from e
```

As-is this causes an error:

```
TypeError('with_traceback() takes exactly one argument (0 given)')
```

**Review by lost-theory — COMMENTED** [review]
Actually I do see one issue from CI. Left a comment.

**Pull request merged**